src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -762,65 +676,63 @@ def is_locked_raw_device(disk_path):
     return 0
 
 
-def get_devices(_sys_block_path='/sys/block', _dev_path='/dev', _mapper_path='/dev/mapper'):
+def get_block_devs_lsblk():
```

Comment:
the removal of the helper functions with this is quite large just so to fix the problem of being independent of execution location. 

This function produces output that:

* ignores non-zero exit status
* does not guard against invalid information in `stdout`
* duplicates the work that `util.lsblk()` already does
* has no unit tests

It is not clear why this is being introduced vs. using `util.lsblk()`

Reply:
1. I didn't ask to _silently eat errors_, but rather to guard against invalid information in `stdout` because when that happens, the user will receive either a traceback or an exception as an error message which will not explain what is going on. 
2. What to test: if the `rc` is non-zero no processing should be done,  that the re.split and list-comprehension is returning the list of lists with 3 items per inner list as described.
3. `lsblk()` does most of what this is doing, including giving the information for `KNAME,NAME,TYPE`. It does it per device and this function does it for all devices. A trivial to understand function doesn't necessarily mean it is a good addition. My problem with this is that it duplicates effort, and increases the surface for problems that existing code has been resilient against by being in production for a couple of years.

Reply:
1. and 2. are addressed.
As for 3. I think trivial but useful functions are great to add. I don't see the benefit of adding another argument or making the device argument optional and adding branches to `util.lsblk` in order to "save" 5 LOC. This additional complexity increases the surface for bugs as well and more so then a small purpose build function imho. Additionallz it makes it harder for a new developer to reason about what the output of a function is.